### PR TITLE
Fix one_hot implementation for Int Tensors (#2500)

### PR DESCRIPTION
**Pull request by maun**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

https://github.com/tracel-ai/burn/issues/2500
Introduced in https://github.com/tracel-ai/burn/pull/2413

### Changes

Fix

### Testing

Ran the example, it worked.

### CODE REVIEW

File: crates/burn-tensor/src/tensor/api/int.rs [hunk 1]
```diff
@@ -53,7 +53,7 @@ where
     pub fn one_hot(self, num_classes: usize) -> Tensor<B, 2, Int> {
         check!(TensorCheck::one_hot_tensor(self.clone(), num_classes));
         let [num_samples] = self.dims();
-        let indices = self.unsqueeze();
+        let indices = self.unsqueeze_dim(1);
         let values = indices.ones_like();
         Tensor::zeros([num_samples, num_samples], &indices.device()).scatter(1, indices, values)
```
Comment (tsanona) [inline crates/burn-tensor/src/tensor/api/int.rs:58]: Hey hey, yes thank you for changing this. Only realized later and was trying push the fix on another PR.
Forget not to also change this :rocket: 
```suggestion
        Tensor::zeros([num_samples, num_classes], &indices.device()).scatter(1, indices, values)
```
Just so I know in the future, where are these tensor operations tested?
Thanks! :star2:

**Comment by maun** [comment]
I don't know, first commit here ;)
Ah, thanks for the fix, had it in my example, but forgot it here.

File: crates/burn-tensor/src/tensor/api/int.rs (hunk 1 above)
Comment (maun) [reply crates/burn-tensor/src/tensor/api/int.rs]: I don't know, first commit here ;) Ah, thanks for the fix, had it in my example, but forgot it here.

**Comment by tsanona** [comment]
Ah right, hopefully the reviewer will answer it. Thanks for changing it!

File: crates/burn-tensor/src/tensor/api/int.rs (hunk 1 above)
Comment (tsanona) [reply crates/burn-tensor/src/tensor/api/int.rs]: Ah right, hopefully the reviewer will answer it. Thanks for changing it!

Comment (agelas) [reply crates/burn-tensor/src/tensor/api/int.rs]: @tsanona Tests for this are in `crates/burn-tensor/src/tests/ops/squeeze.rs` (both squeeze and unsqueeze are in here).

**Review by nathanielsimard — APPROVED** [review]
Thanks for the fix!

**Review by laggui — APPROVED** [review]

**Pull request merged**